discord/interaction.go

```diff
@@ -0,0 +1,129 @@
+package discord
+
+import (
+	"bytes"
+
+	"github.com/diamondburned/arikawa/v3/utils/json"
+)
+
+// https://discord.com/developers/docs/topics/gateway#interactions
+type Interaction struct {
+	ID        InteractionID   `json:"id"`
+	AppID     AppID           `json:"application_id"`
+	Type      InteractionType `json:"type"`
+	Data      InteractionData `json:"data,omitempty"`
+	ChannelID ChannelID       `json:"channel_id,omitempty"`
+	Token     string          `json:"token"`
+	Version   int             `json:"version"`
+	Message   *Message        `json:"message"`
```

Comment:
Message is optional; the tag should indicate that: `json:"message,omitempty"`.

A comment should also be added to clarify when this field is present, or at
least to clarify that it is optional.